screen-budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt

```diff
@@ -45,59 +47,117 @@ class BudgetViewModel @Inject constructor(
     private val baseCurrencyAct: BaseCurrencyAct,
     private val historyTrnsAct: HistoryTrnsAct,
     private val exchangeAct: ExchangeAct
-) : ViewModel() {
+) : ComposeViewModel<BudgetScreenState, BudgetScreenEvent>() {
+
+    private val baseCurrency = mutableStateOf("")
+    private val timeRange = mutableStateOf<FromToTimeRange?>(null)
+    private val budgets = mutableStateOf<ImmutableList<DisplayBudget>>(persistentListOf())
+    private val categories = mutableStateOf<ImmutableList<Category>>(persistentListOf())
+    private val accounts = mutableStateOf<ImmutableList<Account>>(persistentListOf())
+    private val categoryBudgetsTotal = mutableDoubleStateOf(0.0)
+    private val appBudgetMax = mutableDoubleStateOf(0.0)
+    private val reorderModalVisible = mutableStateOf(false)
+    private val budgetModalData = mutableStateOf<BudgetModalData?>(null)
+
+    @Composable
+    override fun uiState(): BudgetScreenState {
+        LaunchedEffect(Unit) {
+            start()
+        }
 
-    private val _timeRange = MutableStateFlow(ivyContext.selectedPeriod.toRange(1))
-    val timeRange = _timeRange.readOnly()
+        return BudgetScreenState(
+            baseCurrency = getBaseCurrency(),
+            categories = getCategories(),
+            accounts = getAccounts(),
+            budgets = getBudgets(),
+            categoryBudgetsTotal = getCategoryBudgetsTotal(),
+            appBudgetMax = getAppBudgetMax(),
+            timeRange = getTimeRange(),
+            reorderModalVisible = getReorderModalVisible(),
+            budgetModalData = getBudgetModalData()
+        )
+    }
 
-    private val _baseCurrencyCode =
-        MutableStateFlow(com.ivy.legacy.utils.getDefaultFIATCurrency().currencyCode)
-    val baseCurrencyCode = _baseCurrencyCode.readOnly()
+    override fun onEvent(event: BudgetScreenEvent) {
```

Comment:
Can you place `onEvent` below the `@Composable fun getSomething` functions?

Reply:
Your wish has been granted 😄